### PR TITLE
Move virtual engine processor to GridDevice

### PR DESCRIPTION
**Pull request by verult**
Moves the various device-related functions in `virtual_engine_processor.py` to GridDevice. This is blocking SerializableDevice deprecation (#5522) because library code cannot call the deprecated SerializableDevice.

* Rerouted functions in `virtual_engine_processor.py` to `GridDevice`
* Deprecated the `gate_sets` parameter in `virtual_engine_processor.py`
* Move `create_device_from_processor_id()` to GridDevice. @95-martin-orion 
  * Made the return type more general by changing it to `cirq.Device` to allow for new device implementations in the future. 
* Created copies of existing rainbow and weber DeviceSpecifications and changed them to match the new DeviceSpecification format.
  * This update needs to happen now because `create_noiseless_virtual_engine_from_templates()` is updated to return a `GridDevice`, and the templates (rainbow and weber DeviceSpecifications) need to contain data in the new format in order to parse into GridDevice properly.
  * There are no publicly available devices since weber, so I chose to repurpose the existing rainbow and weber information. This was done manually since these devices are no longer available.
    * Gateset and gate_duration information should match previous versions. In particular, they both include the CZ gate with no duration information.

**Reviewer Note**: The first 3 commits are from another pending PR (#5558); please start with the commit "[Move virtual engine functions to GridDevice](https://github.com/quantumlib/Cirq/pull/5561/commits/2e9ef33897f926f5b29481cb5b5f6192d814d7ba)". Don't be discouraged by the line count :) other than the 3 borrowed commits, ~600 lines are from the two new DeviceSpecification messages.

@mpharrigan 
cc @MichaelBroughton @dstrain115

### CODE REVIEW

**Review by MichaelBroughton — APPROVED** [review]
LGTM will leave to @dstrain115 for a quick check off.

File: cirq-google/cirq_google/engine/engine_processor_test.py [hunk 1]
```diff
@@ -291,6 +291,7 @@ def test_supported_languages():
 
 
 def test_get_device_specification():
+    # TODO(verult)
     processor = cg.EngineProcessor('a', 'p', EngineContext(), _processor=quantum.QuantumProcessor())
     assert processor.get_device_specification() is None
```
Comment (dstrain115) [inline cirq-google/cirq_google/engine/engine_processor_test.py:296]: Why is this None?  Is this an incomplete test?

**Review by dstrain115 — APPROVED** [review]
Looks fine to me besides the unfinished test.

File: cirq-google/cirq_google/engine/engine_processor_test.py (hunk 1 above)
Comment (verult) [reply cirq-google/cirq_google/engine/engine_processor_test.py:296]: This was here from the previous iteration of the test. I think this part just verifies that an unset DeviceSpecification doesn't break anything. The next section verifies an actual DeviceSpecification.

File: cirq-google/cirq_google/engine/engine_processor_test.py [hunk 2]
```diff
@@ -291,6 +291,7 @@ def test_supported_languages():
 
 
 def test_get_device_specification():
+    # TODO(verult)
```
[the remark this answers is not in the document]
Comment (verult) [reply cirq-google/cirq_google/engine/engine_processor_test.py:294]: Oops added as a reminder to come back and update the test but forgot to remove it after that. Thanks!

**Pull request merged**